### PR TITLE
Add Fourier-MIONet paper

### PR DESCRIPTION
**Pull request by Jerry-Jzy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ DeepXDE is a library for scientific machine learning and physics-informed learni
     - multifidelity DeepONet [[Phys. Rev. Research](https://doi.org/10.1103/PhysRevResearch.4.023210)]
     - DeepM&Mnet: solving multiphysics and multiscale problems [[J. Comput. Phys.](https://doi.org/10.1016/j.jcp.2021.110296), [J. Comput. Phys.](https://doi.org/10.1016/j.jcp.2021.110698)]
     - Reliable extrapolation [[arXiv](https://arxiv.org/abs/2212.06347)]
+    - Fourier-MIONet [[arXiv](https://arxiv.org/abs/2303.04778)]
```
Comment (lululxvi) [inline README.md:34]: Move Fourier-MIONet after MIONet

Comment (Jerry-Jzy) [reply README.md]: Fixed

File: docs/user/research.rst [hunk 2]
```diff
@@ -200,6 +200,7 @@ PINN
 DeepONet
 --------
 
+#. Z\. Jiang, M. Zhu, D. Li, Q. Li, Y. Yuan, & L. Lu. `Fourier-MIONet: Fourier-enhanced multiple-input neural operators for multiphase modeling of geological carbon sequestration <https://arxiv.org/pdf/2303.04778.pdf>`_. *arXiv preprint arXiv:2303.04778*, 2023.
```
[the remark this answers is not in the document]
Comment (Jerry-Jzy) [reply docs/user/research.rst:205]: Fixed

File: docs/user/research.rst [hunk 3]
```diff
@@ -103,7 +104,8 @@ DeepXDE has been used in
   `BirenTech Research <https://www.birentech.com>`_,
   `General Motors <https://www.gm.com>`_,
   `RocketML <https://www.rocketml.net>`_,
-  `Saudi Aramco <https://www.aramco.com>`_
+  `Saudi Aramco <https://www.aramco.com>`_,
+  `ExxonMobil <https://corporate.exxonmobil.com>`_
```
Comment (lululxvi) [inline docs/user/research.rst:108]: Move ExxonMobil after BirenTech Research

Comment (Jerry-Jzy) [reply docs/user/research.rst]: Fixed

File: docs/user/research.rst [hunk 4]
```diff
@@ -42,6 +42,7 @@ DeepXDE has been used in
   `University of Electronic Science and Technology of China <https://en.uestc.edu.cn>`_,
   `KTH Royal Institute of Technology <https://www.kth.se/en>`_,
   `Florida State University <https://www.fsu.edu>`_,
+  `University of Delaware <https://www.udel.edu>`_,
```
Comment (lululxvi) [inline docs/user/research.rst:45]: this should be sorted by US news ranking https://www.usnews.com/education/best-global-universities/rankings

Comment (Jerry-Jzy) [reply docs/user/research.rst]: Fixed

**Pull request merged**